packages/flutter_tools/test/integration.shard/web_plugin_registrant_test.dart

```diff
@@ -41,6 +41,35 @@ void main() {
     await _restoreFlutterToolsSnapshot();
   });
 
+  // See: https://github.com/dart-lang/dart-services/pull/874
+  testUsingContext('generated plugin registrant for dartpad exists and passes analysis', () async {
+    await _createProject(projectDir, <String>[]);
+    // We need to add a dependency with web support to trigger
+    // the generated_plugin_registrant generation.
+    await _addDependency(projectDir, 'shared_preferences',
+        version: '^2.0.0');
+    // The plugin registrant for dartpad is created on flutter pub get.
+    await _doFlutterPubGet(projectDir);
+
+    final Directory dartpadDir = projectDir.childDirectory('.dart_tool/dartpad');
+    expect(dartpadDir.childFile('web_plugin_registrant.dart'), exists);
+
+    await _analyzeEntity(dartpadDir.childFile('web_plugin_registrant.dart'));
```

Comment:
Do we already test if `registerPlugins()` inside `web_plugin_registrant.dart` is empty when the project do not have any plugins and the other way around?

Reply:
Let me check, I don't remember if this was tested already, or we need to check the contents of the file!

Reply:
Nope, I thought this'd happen in an unit test elsewhere, but there we're checking the file that *imports* the plugin_registrant, not the registrant itself. I've updated the tests to look at the contents of the generated file.